### DISCUSSION TITLE
VNC over ssh

### DISCUSSION THREAD

**Discussion by mariomadproductions** [gujjwal00/avnc#176] [General]
Is there a way to use VNC over SSH:
1) without storing the ssh key password in the settings?
or
2) over an ssh tunnel created by termux?

**Comment by gujjwal00** [comment]
> Is there a way to use VNC over SSH:
> 
>     1. without storing the ssh key password in the settings?

Yes, you can leave the key password field blank. It was recently made optional, see https://github.com/gujjwal00/avnc/issues/131 . In fact the password field will be removed entirely in a future version.

> 
>     2. over an ssh tunnel created by termux?

Yes, use `host:port` of the tunnel in AVNC, instead of using the address of VNC server directly.

**Reply by mariomadproductions** [reply]
Thanks, that change should work. Although I couldn't get it to work over port forwarding with termux. But using it directly in the app should be better anyway.

**Reply by zaknick** [reply]
I use termux to ssh and port forward 

ssh -L 5901:127.0.0.1:5901 -C -N -l (username) xxx.xxx.xxx.xxx (host IP)

then avnc

127.0.0.1:5901

works fine

**Comment by pcrow** [comment]
If there's a way to launch a particular avnc connection from the Termux command-line, then you could have a script that set up tunnels as needed before launching avnc, and you could run it all as a Termux widget once it was set up.
